Issue 473: change how output directory is generated for extract_xri script

rminsil opened the pull request:
This PR addresses one point raised in the code review feedback from Damien from this comment:

https://github.com/sillsdev/silnlp/pull/491#pullrequestreview-2261754354

> I would make the output directory an optional command-line argument. By default, you should extract files to `SIL_NLP_ENV.mt_corpora_dir`. This should point to the MT/corpora folder in the S3 bucket, which is where we would store this kind of corpus.

I wasn't clear on the directory structure to use within the corpora dir, so I kept the existing logic of generating a unique folder based on the cli inputs + timestamp.

Note that this PR is built off the back of https://github.com/sillsdev/silnlp/pull/506. When that one is merged I'll rebase this one and have it target `master`.

Once this PR is merged, all code review comments from #491 are addressed and balance is returned to the force.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/silnlp/507)
<!-- Reviewable:end -->

ddaspit reviewed (changes requested):
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/silnlp/507)* status: all files reviewed, 1 unresolved discussion (waiting on @rminsil)

___
*[`silnlp/common/extract_xri.py` line 116 at r1](https://reviewable.io/reviews/sillsdev/silnlp/507#-O63EXxr5YGvzPLwsXaw:-O63EXxr5YGvzPLwsXax:b-fn9zax) ([raw file](https://github.com/sillsdev/silnlp/blob/2477b3cb90d7b06d3943165cd8d113d4d6805026/silnlp/common/extract_xri.py#L116)):*
> ```Python
>     if cli_input.output is None:
>         unique_dir = f"{cli_input.source_iso}-{cli_input.target_iso}-{cli_input.dataset_descriptor}-{time.strftime('%Y%m%d-%H%M%S')}"
>         output_dir = Path(os.path.join(SIL_NLP_ENV.mt_corpora_dir, unique_dir))
> ```

`SIL_NLP_ENV.mt_corpora_dir` is already a `Path`, so you should be able to do:

``` py
output_dir = SIL_NLP_ENV.mt_corpora_dir / unique_dir
```


<!-- Sent from Reviewable.io -->

ddaspit reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/silnlp/507#-:-O6bzst2BgxVDgI7BxSb:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @rminsil)



<!-- Sent from Reviewable.io -->

rminsil commented:
@ddaspit I found the logging config in the parent __init.py__ and understand what you mean now.

I've removed the option to log to a file.

I've changed the logger name to be specific to the module. I'll make an analogous change to the repair logger used in #515.

I've added an optional flag to allow setting the debug level from the command line.
